src/widget/systemtrayicon.cpp

```diff
@@ -116,6 +113,26 @@ QString SystemTrayIcon::extractIconToFile(QIcon icon, QString name)
     return iconPath;
 }
 
+#if defined(ENABLE_SYSTRAY_GTK_BACKEND) || defined(ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND)
+GdkPixbuf* SystemTrayIcon::convertQIconToPixbuf(const QIcon &icon)
+{
+    void (*callbackFreeImage)(guchar*, gpointer) =
+        [](guchar* image_bytes, gpointer)
+    {
+        delete[] image_bytes;
```

Comment:
This works, GdkPixbuf passes the bytes to free as first argument, the other is for any possible additional data required. (see https://developer.gnome.org/gdk-pixbuf/unstable/gdk-pixbuf-Image-Data-in-Memory.html#gdk-pixbuf-new-from-data)

And if you like it this way, I'll keep it, because for now I'm not quite sure how to keep the image around, as I'm a C++ noob and `QPixmap::toImage()` doesn't return a pointer, so I'd somewhat either need to trick the C++ code to understand the C callback took a reference, or copy the QImage, which, unless it implements clever buffer sharing, is likely to involve the same copy.